src/lib/error/bad-data-error.ts

```diff
@@ -117,12 +118,15 @@ const enumMessage = (
 };
 
 export const fromOpenApiValidationError =
-    (request: { body: object; query: object }) =>
+    (data: object) =>
     (validationError: ErrorObject): ValidationErrorDescription => {
         const { instancePath, params, message } = validationError;
         const [errorSource, substringOffset] = instancePath.startsWith('/body')
-            ? [request.body, '/body/'.length]
-            : [request.query, '/query/'.length];
+            ? [(data as { body: object }).body, '/body/'.length]
+            : instancePath.startsWith('/query')
+              ? [(data as { query: object }).query, '/query/'.length]
+              : // This is not validating body or query, so we can assume this is manually validating plain data
+                [data, 1];
```

Comment:
I'm not suuuuper hot on this implementation, to be frank. One thing is that you'll lose autocompletion, but it feels a little loose when you can pass any object. You could send an empty object, but put `/body` in the instance path and it'd blow up. But then again, this is typescript. You can lie to it as much as you want 🤷🏼 

But I'm not gonna die on that hill, if the rest of you think this is a better implementation.

Reply:
Hey @thomasheartman, I did a few more changes to make this even more dynamic. Let me know what you think.

>One thing is that you'll lose autocompletion, but it feels a little loose when you can pass any object.

Looking at our code I don't think we're taking advantage of the current type anyways. I agree it feels loose, but I also think it makes sense, if we're allowing you to validate anything against a schema. Why should we be limited to objects with `body` and `query` properties?

>You could send an empty object, but put /body in the instance path and it'd blow up.

I don't think it will (with the recent changes), and I added a test that validates that.

>But I'm not gonna die on that hill, if the rest of you think this is a better implementation.

I'm also open to alternatives, if you think we can do this in a better way!